packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts

```diff
@@ -0,0 +1,86 @@
+//// { order: 3, compiler: { strictNullChecks: false } }
+
+// No JavaScript existem duas maneiras de declarar valores que não
+// existem, o TypesScript adiciona sintaxe extra que nos proporciona
+// mais maneiras de declarar algo como opicional ou nulo.
+
+// Primeiramente a diferença entre os dois primitivos
+// no JavaScript: undefined e null.
+
+// Undefined é quando algo não foi encontrado ou definido.
+
+const emptyObj = {};
+const anUndefinedProperty: undefined = emptyObj["anything"];
+
+// Null é usado quando se quer intencionalmente mostrar que
+// o valor não existe.
+
+const searchResults = {
+  video: { name: "LEGO Movie" },
+  text: null,
+  audio: { name: "LEGO Movie Soundtrack" },
+};
+
+// Por que não usar undefined? Pricipalmente, porque agora podemos
+// verificar que text foi incluido corretamente. Se text retornasse
+// undefined então o resultado seria o mesmo que dizer que não
+// foi declarado.
+
+// Isso pode parecer superficial, mas quando convertido
+// em uma JSON string, se o text fosse undefined, nāo seria
+// incluido na conversão para string.
+
+// Strict Null Types
+
+// Antes do TypeScrips 2.0, undefined e null eram efetivamente
+// ignorados no sistema de typagem. Isso fazia com que TypeScript
+// pudesse prover um ambiente de codagem mais próximo de um JavaScript
+// sem tipagem.
+
+// A versāo 2.0 adiciona um marcador de compilagem, chamado "strictNullChecks"
+// esse marcador requer que undefined e null fossem tratados como tipos,
+// o que significa que devem ser manipulados via análise de code-flow.
+// ( veja mais em example:code-flow )
```

Comment:
```suggestion
// A versāo 2.0 adiciona uma configuração de compilação (flag),
// chamada "strictNullChecks". Essa flag requer que undefined e null
// sejam tratados como tipos, o que significa que devem ser manipulados
// via análise de fluxo de código (code-flow).
// ( veja mais em example:code-flow )
```

Reply:
Yes, some words are harder to find a good translation for. Thanks